### PR TITLE
Fixed api.yaml.ejs so it does not connect to Zalando

### LINKED ISSUE #21445: Do not link to third-party Zalando website
Hello,

this reference to the Zalando website:

https://github.com/jhipster/generator-jhipster/blob/7dd8197465cf71abda38f260f376ac8bff9670c3/generators/server/templates/src/main/resources/swagger/api.yml.ejs#L38

may cause breakage on some setups, especially because Zalando might be blacklisted in some networks (to prevent employees from shopping on duty). It may seem exaggerated but I've seen it break a pipeline with my own eyes.

### PR DESCRIPTION
**Pull request by fpesari**
Replaced the reference to https://opensource.zalando.com/problem/schema.yaml#/Problem with the corresponding YAML data. Fixes #21445

### CODE REVIEW

File: generators/server/templates/src/main/resources/swagger/api.yml.ejs [hunk 1]
```diff
@@ -35,7 +35,46 @@ components:
...
+              properties:
+                type:
+                  type: string
+                  format: uri
+                  description: |
+                    An absolute URI that identifies the problem type.  When dereferenced,
+                    it SHOULD provide human-readable documentation for the problem type
+                    (e.g., using HTML).
+                  default: 'about:blank'
+                  example: 'https://zalando.github.io/problem/constraint-violation'
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/swagger/api.yml.ejs:49]:
```suggestion
                  example: 'https://www.jhipster.tech/problem/constraint-violation'
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/resources/swagger/api.yml.ejs [hunk 2]
```diff
@@ -35,7 +35,46 @@ components:
       content:
         "application/problem+json":
           schema:
-            $ref: 'https://opensource.zalando.com/problem/schema.yaml#/Problem'
+            Problem:
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/swagger/api.yml.ejs:38]: This line needs to be removed and review indentation

Comment (fpesari) [reply generators/server/templates/src/main/resources/swagger/api.yml.ejs:38]: Hello, do you mean remove the name of the object (Problem) and de-indent everything by one level?

Comment (DanielFran) [reply generators/server/templates/src/main/resources/swagger/api.yml.ejs:38]: Yes. But apply first the 2 changes requested, and if it fails try this, please

Comment (fpesari) [reply generators/server/templates/src/main/resources/swagger/api.yml.ejs:38]: Should I do it in new commits or amend this?

Comment (DanielFran) [reply generators/server/templates/src/main/resources/swagger/api.yml.ejs:38]: Like you want 🙂

**Comment by fpesari** [comment]
@DanielFran I implemented your first two suggestions. I'll try the third one if CI fails.

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
Thanks @fpesari for the contribution

**Pull request merged**